cli/start.go

```diff
@@ -440,10 +445,17 @@ func runStart(_ *cobra.Command, args []string) error {
 	}()
 
 	select {
-	case <-signalCh:
-		log.Errorf(context.TODO(), "second signal received, initiating hard shutdown")
+	case sig := <-signalCh:
+		err := fmt.Errorf("received signal '%s' during shutdown, initiating hard shutdown", sig)
+		log.Errorf(context.TODO(), "%v", err)
+		// On Unix, os.Signal is syscall.Signal and it's convertible to int.
+		ErrorCode = 128 + int(sig.(syscall.Signal))
+		return err
 	case <-time.After(time.Minute):
-		log.Errorf(context.TODO(), "time limit reached, initiating hard shutdown")
+		err := errors.New("time limit reached, initiating hard shutdown")
+		log.Errorf(context.TODO(), "%v", err)
+		ErrorCode = 2
```

Comment:
OK, fair enough. I'd still lean towards never returning exit status 0 even on clean shutdown, but I don't feel strongly enough about it.